File: evidence/generated-systems/golang/aerospace-on-golang/aerospace-on-golang/internal/model/Supplier.go
package model

import (
    "gorm.io/gorm"
)

//==============================================================
// Supplier Declaration
//==============================================================
type Supplier struct {
    gorm.Model
     Name                                    string
     Manufacturers           []AerospaceManufacturer `gorm:"foreignKey:ManufacturersFromSupplierId"`
     Components           []Component_ `gorm:"foreignKey:ComponentsFromSupplierId"`
     EngineTypes           []EngineType `gorm:"foreignKey:EngineTypesFromSupplierId"`
     AvionicsSuites           []AvionicsSuite `gorm:"foreignKey:AvionicsSuitesFromSupplierId"`
     Apus           []APU `gorm:"foreignKey:ApusFromSupplierId"`
     LandingGears           []LandingGear `gorm:"foreignKey:LandingGearsFromSupplierId"`
    SupplierType                      SupplierType
    ApprovalStatus                      SupplierApprovalStatus

// parent associations as their child

}

File: evidence/generated-systems/rubyonrails/aerospace-on-rails/aerospaceonrails/app/models/supplier.rb
class Supplier < ApplicationRecord
  enum SupplierType: [:Airframe, :Engine, :Avionics, :Systems, :Materials, :MRO, :Testing]
  enum ApprovalStatus: [:Applied, :Approved, :OnHold, :Suspended]


  has_many :Manufacturers, class_name: 'AerospaceManufacturer'
  has_many :Components, class_name: 'Component_'
  has_many :EngineTypes, class_name: 'EngineType'
  has_many :AvionicsSuites, class_name: 'AvionicsSuite'
  has_many :Apus, class_name: 'APU'
  has_many :LandingGears, class_name: 'LandingGear'

end
